src/videojs-fairplay.js

```diff
@@ -0,0 +1,239 @@
+    }, false);
+
+    request.open('GET', certificateUrl, true);
+    request.send();
+  }
+
+  fetchLicense({ target, message }) {
+    const { licenseUrl } = this.protection_;
+
+    const request = new XMLHttpRequest();
+
+    request.responseType = 'arraybuffer';
+    request.session = target;
+
+    request.addEventListener('error', this.onLicenseError, false);
+    request.addEventListener('load', this.onLicenseLoad, false);
+
+    request.open('POST', licenseUrl, true);
+    request.setRequestHeader('Content-type', 'application/octet-stream');
+    request.send(message);
+  }
+
+  hasProtection({ certificateUrl, keySystem, licenseUrl } = {}) {
+    return certificateUrl && keySystem && licenseUrl;
+  }
+
+  log(message) {
+    if (!logToBrowserConsole) {
+      return;
+    }
+
+    console.log(message);
+  }
+
+  onCertificateError() {
+    this.player_.error({
+      code: 0,
+      message: 'Failed to retrieve the server certificate.',
+    });
+  }
+
+  onCertificateLoad(event, { callback }) {
+    this.certificate = new Uint8Array(event.target.response);
+
+    this.el_.addEventListener('error', this.onVideoError, false);
+    this.el_.addEventListener('webkitneedkey', this.onVideoWebkitNeedKey, false);
+
+    callback();
+  }
+
+  onKeySessionWebkitKeyAdded() {
+    this.log('Decryption key was added to the session.');
+  }
+
+  onKeySessionWebkitKeyError() {
+    this.player_.error({
+      code: 0,
+      message: 'A decryption key error was encountered.',
+    });
+  }
+
+  onKeySessionWebkitKeyMessage(event) {
+    const message = event.message;
+    const target = event.target;
+
+    this.fetchLicense({
+      message,
+      target,
+    });
+  }
+
+  onLicenseError() {
+    this.player_.error({
+      code: 0,
+      message: 'The license request failed.',
+    });
+  }
+
+  onLicenseLoad(event) {
+    event.target.session.update(new Uint8Array(event.target.response));
+  }
+
+  onVideoError() {
+    this.player_.error({
+      code: 0,
+      message: 'A video playback error occurred.',
+    });
+  }
+
+  onVideoWebkitNeedKey(event) {
+    const { keySystem } = this.protection_;
+
+    const contentId = getHostnameFromURI(arrayToString(event.initData));
+
+    const initData = concatInitDataIdAndCertificate(event.initData, contentId, this.certificate);
+
+    const keySession = this.createKeySession(keySystem, initData);
+
+    keySession.contentId = contentId;
+
+    keySession.addEventListener('webkitkeyadded', this.onKeySessionWebkitKeyAdded, false);
+    keySession.addEventListener('webkitkeyerror', this.onKeySessionWebkitKeyError, false);
+    keySession.addEventListener('webkitkeymessage', this.onKeySessionWebkitKeyMessage, false);
+  }
+
+  src({ src }) {
+    if (!this.hasProtection(this.protection_)) {
+      return this.tech_.src(src);
+    }
+
+    return this.fetchCertificate({
+      callback: () => {
+        this.tech_.src(src);
+      },
+    });
+  }
+
+  static setLogToBrowserConsole(value = false) {
```

Comment:
I think this is only a requirement of `eslint-plugin-react`? Will move it up anyways.